src/express2.ml

```diff
@@ -0,0 +1,172 @@
+type done_
+(** abstract type which ensure middleware function must either
+    call the [next] function or one of the [send] function on the 
+    response object. 
+
+    This should be a great argument for OCaml, the type system 
+    prevents silly error which in this case would make the server hang *)
+
+(** TODO : maybe this should be a more common module like bs-error *)
+module Error = struct 
+
+  type t 
+
+  external message : t -> string option = "" 
+    [@@bs.send] [@@bs.return null_undefined_to_opt]
+  
+  external name : t -> string option = "" 
+    [@@bs.send] [@@bs.return null_undefined_to_opt]
+
+end 
+
+module Request = struct 
+
+  type t 
+
+  type params = Js_json.t Js_dict.t
+
+  external params : t -> params = "" [@@bs.get]
+end 
+
+module Response = struct 
+
+  type t 
+
+  external sendString : t -> string -> done_ = "send" [@@bs.send] 
```

Comment:
Note here that the return type is `done_` not `unit`. This is to enforce that either `next()` or one of the `send` method is called.